engine/src/main/java/org/pentaho/di/trans/steps/rssoutput/RssOutput.java

```diff
@@ -0,0 +1,763 @@
+          OutputFormat format = org.dom4j.io.OutputFormat.createPrettyPrint();
+          // Set encoding ...
+          if ( Utils.isEmpty( meta.getEncoding() ) ) {
+            format.setEncoding( "iso-8859-1" );
+          } else {
+            format.setEncoding( meta.getEncoding() );
+          }
+
+          try {
+            XMLWriter writer = new XMLWriter( new FileWriter( new File( data.filename ) ), format );
+            writer.write( data.document );
+            writer.close();
+          } catch ( Exception e ) {
+            // Ignore errors
+          } finally {
+            data.document = null;
+          }
+
+        }
+      }
+      setOutputDone();
+      return false;
+    }
+
+    if ( first ) {
+      first = false;
+      data.inputRowMeta = getInputRowMeta();
+      data.outputRowMeta = data.inputRowMeta.clone();
+      meta.getFields( data.outputRowMeta, getStepname(), null, null, this, repository, metaStore );
+      // Let's check for filename...
+
+      if ( meta.isFilenameInField() ) {
+        if ( Utils.isEmpty( meta.getFileNameField() ) ) {
+          logError( BaseMessages.getString( PKG, "RssOutput.Log.FilenameFieldMissing" ) );
+          setErrors( 1 );
+          stopAll();
+          return false;
+        }
+
+        // get filename field index
+        data.indexOfFieldfilename = data.inputRowMeta.indexOfValue( meta.getFileNameField() );
+        if ( data.indexOfFieldfilename < 0 ) {
+          // The field is unreachable !
+          logError( BaseMessages.getString( PKG, "RssOutput.Log.ErrorFindingField", meta.getFileNameField() ) );
+          throw new KettleException( BaseMessages.getString( PKG, "RssOutput.Log.ErrorFindingField", meta
+            .getFileNameField() ) );
+        }
+
+      } else {
+        data.filename = buildFilename();
+      }
+
+      // Check if filename is empty..
+      if ( Utils.isEmpty( data.filename ) ) {
+        logError( BaseMessages.getString( PKG, "RssOutput.Log.FilenameEmpty" ) );
+        throw new KettleException( BaseMessages.getString( PKG, "RssOutput.Log.FilenameEmpty" ) );
+      }
+
+      // Do we need to create parent folder ?
+      if ( meta.isCreateParentFolder() ) {
+        // Check for parent folder
+        FileObject parentfolder = null;
+        try {
+          // Get parent folder
+          parentfolder = KettleVFS.getFileObject( data.filename, getTransMeta() ).getParent();
+          if ( !parentfolder.exists() ) {
+            if ( log.isDetailed() ) {
+              logDetailed( BaseMessages.getString( PKG, "RssOutput.Log.ParentFolderExists", parentfolder
+                .getName().toString() ) );
+            }
+            parentfolder.createFolder();
+            if ( log.isDetailed() ) {
+              logDetailed( BaseMessages.getString( PKG, "RssOutput.Log.CanNotCreateParentFolder", parentfolder
+                .getName().toString() ) );
+            }
+          }
+        } catch ( Exception e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 186.